### ISSUE TITLE
Refactor asset terminology to tokens

### ISSUE THREAD

**Issue by PaulRBerg** [sablier-labs/v2-core#991]
To align the NatSpec with the terminology used in our PR and documentation site, we should refactor all references to 'assets' to say 'tokens' instead.

**Comment by smol-ninja** [comment]
I also like calling them tokens over assets because the asset is a broad category whereas token has a specific meaning. 

But, I remember having a conversation during one of the discussions / PR review / on slack with @andreivladbrg. IIRC, the reason that the "assets" terminology was chosen was because the "token" was reserved to represent the NFT (as ERC721 "token"). Has that opinion be changed?

**Comment by PaulRBerg** [comment]
> the reason that the "assets" terminology was chosen was because the "token" was reserved to represent the NFT

Correct, that was the original rationale in ~2022. But then, with time, we realized that the 'token' terminology is ubiquitous for ERC-20s. So much so that the slight confusion that may occur due to the NFT also being referred to as a 'token' is negligible compared to the inaccuracy of referring to ERC-20s as assets.

> Has that opinion be changed?

Yes. I can't find it now, but we discussed it on Slack.

**Comment by PaulRBerg** [comment]
Are we still on track to perform this rename in v1.3.0?

**Comment by smol-ninja** [comment]
Yes.

**Comment by smol-ninja** [comment]
Good point. My take is that we use `streamID` to refer to ERC721's token ID instead of `tokenId` in our telegram chats as well as in the docs. Sablier streams are represented by NFTs. I understand ERC721 standard uses `tokenID` but we don't use that name anywhere or do we? 

Referring to EIPs, EIP20 does not use the word `asset` as well. So if we are referring to EIPs, crypto users seem to be familiar with `token` while referring to ERC20 and `tokenID` while referring to ERC721.

TL;DR we can use `token` for ERC20 and `streamID` for stream ID.

**Comment by PaulRBerg** [comment]
I am massively in favor of using tokens for referring to ERC-20s. We also discussed this on Slack, and not just about MNTs (maybe @maxdesalle can dig up the conversation, I can't find it now).

`tokenId` is a property that only some users see. Whereas all users see `asset`. I wouldn't name the latter property based on the former.

Also, nobody refers to NFTs as tokens, despite the eponymous reference in the acronym.

Plus, what Shub said.

**Comment by razgraf** [comment]
Fair points from both! Let's go with `token` then.
